caffe2/core/dispatch/DispatchKey.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "caffe2/core/dispatch/DeviceId.h"
+#include "caffe2/core/dispatch/LayoutId.h"
+#include "caffe2/core/typeid.h"
+
+#include <vector>
+#include <functional>
+#include "caffe2/utils/Array.h"
+
+namespace c10 {
+
+namespace details {
+struct TensorParameterDispatchKey final {
```

Comment:
Can we have a comment saying that the parameter dispatch key internal structure is not finalized, and so you should not rely on the internal representation of it? Similarly, I'd feel much better if the fields were private and we used friends to prevent people from accessing them directly, so that you have to stop and think if you want to rely on something inside.